include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolation/BarycentricRational.hpp

```diff
@@ -0,0 +1,112 @@
+
+/// @project        Open Space Toolkit ▸ Mathematics
+/// @file           OpenSpaceToolkit/Mathematics/CurveFitting/7Interpolation/BarycentricRational.hpp
+/// @author         Vishwa Shah <vishwa@loftorbital.com
+/// @license        Apache License 2.0
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef __OpenSpaceToolkit_Mathematics_Interpolator_BarycentricRational__
+#define __OpenSpaceToolkit_Mathematics_Interpolator_BarycentricRational__
+
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+
+#include <boost/math/interpolators/barycentric_rational.hpp>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace ostk
+{
+namespace math
+{
+namespace curvefitting
+{
+namespace interp
+{
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+using ostk::core::types::Real ;
+using ostk::core::types::Size ;
+
+using ostk::math::obj::VectorXd ;
+
+using boost::math::interpolators::barycentric_rational ;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @brief                      BarycentricRational
+///
+///                             Barycentric rational interpolation is a high-accuracy interpolation method for non-uniformly spaced samples. It requires 𝑶(N) time for construction, and 𝑶(N) time for each evaluation.
+///
+/// @ref                        https://www.boost.org/doc/libs/1_81_0/libs/math/doc/html/math_toolkit/barycentric.html
+
+class BarycentricRational
+{
+
+    public:
+
+        /// @brief              Constructor
+        ///
+        /// @code
+        ///                     BarycentricRational barycentricRational(x, y) ;
+        /// @endcode
+        ///
+        /// @param              [in] x A vector of x values
+        /// @param              [in] y A vector of y values
+        ///
+        /// @warning            The x values must be sorted in ascending order
+        /// @warning            The x values must be equally spaced
+
+                                BarycentricRational                                 (   const   VectorXd&                   x,
+                                                                                        const   VectorXd&                   y                                           ) ;
+
+        /// @brief              Clone barycentric rational
+        ///
+        /// @return             Pointer to cloned barycentric rational
+
+        BarycentricRational*    clone                                       ( ) const ;
+
```

Comment:
Since this class doesn't inherit from another class, there's no need for a cloning pattern:

```suggestion
```

Reply:
Good point.

On that note though what do you think about having a parent Interpolator class that has an evaluate method?
That way maybe in astro in the Tabulated class we can have a pointer to an interpolator. I'll play with it tomorrow and see what I come up with.

Reply:
Added a parent class so going to leave the clone as is.

Reply:
Awesome!